nemo/collections/nlp/modules/common/bert_module.py

```diff
@@ -48,16 +48,17 @@ def save_to(self, save_path: str):
         """Saves module/model with weights"""
         pass
 
-    def restore_weights(self, restore_path: str, prefix='bert.'):
+    def restore_weights(self, restore_path: str, prefix='bert_model.'):
         """Restores module/model's weights"""
         logging.info(f"restore from {restore_path}")
-        pretrained_dict = torch.load(restore_path)
+        pretrained_dict = torch.load(restore_path)["state_dict"]
 
         # remove prefix from pretrained dict
         if prefix in list(pretrained_dict.keys())[0]:
             pretrained_dict = {k[len(prefix) :]: v for k, v in pretrained_dict.items()}
 
         model_dict = self.state_dict()
         pretrained_dict = {k: v for k, v in pretrained_dict.items() if k in model_dict}
+        assert len(pretrained_dict) == len(model_dict)
         model_dict.update(pretrained_dict)
         self.load_state_dict(model_dict)
```

Comment:
https://github.com/NVIDIA/NeMo/blob/token_class_patch/nemo/collections/nlp/modules/common/bert_module.py#L51  This version works with the old BERT checkpoints and get_pretrained_lm() if bert_checkpoint is specified in the config . I think it's better to keep 'bert.' prefix as the default to avoid running conversion script (the new BERT encoder checkpoint should still work).